src/main/job-queues/redlocked/s3-queue.ts

```diff
@@ -0,0 +1,180 @@
+import { S3 } from 'aws-sdk'
+import { randomBytes } from 'crypto'
+import { DateTime } from 'luxon'
+import { gunzipSync, gzipSync } from 'zlib'
+
+import { EnqueuedJob, PluginsServerConfig } from '../../../types'
+import { S3Wrapper } from '../../../utils/db/s3-wrapper'
+import { UUIDT } from '../../../utils/utils'
+import { JobQueueBase } from '../job-queue-base'
+
+const S3_POLL_INTERVAL = 5000
+
+export class S3Queue extends JobQueueBase {
+    serverConfig: PluginsServerConfig
+    s3Wrapper: S3Wrapper | null
+    runner: NodeJS.Timeout | null
+
+    constructor(serverConfig: PluginsServerConfig) {
+        super()
+        this.serverConfig = serverConfig
+        this.s3Wrapper = null
+        this.runner = null
+    }
+
+    // producer
+
+    async connectProducer(): Promise<void> {
+        await this.connectS3()
+    }
+
+    async enqueue(retry: EnqueuedJob): Promise<void> {
+        if (!this.s3Wrapper) {
+            throw new Error('S3 object not initialized')
+        }
+        const date = new Date(retry.timestamp).toISOString()
+        const [day, time] = date.split('T')
+        const dayTime = `${day.split('-').join('')}-${time.split(':').join('')}`
```

Comment:
was there a specific reason for formatting things like this?

Reply:
Basically just keeping it sortable to the second while removing special symbols that might interfere with it being a filename.